src/test/java/org/jabref/logic/cleanup/CleanupWorkerTest.java

```diff
@@ -229,117 +238,121 @@ public void cleanupMoveFilesMovesFileFromSubfolder() throws IOException {
 
         worker.cleanup(preset, entry);
         LinkedFile newFileField = new LinkedFile("", tempFile.getName(), "");
-        Assert.assertEquals(Optional.of(FileFieldWriter.getStringRepresentation(newFileField)), entry.getField("file"));
+        assertEquals(Optional.of(FileFieldWriter.getStringRepresentation(newFileField)), entry.getField("file"));
     }
 
     @Test
-    public void cleanupRelativePathsConvertAbsoluteToRelativePath() throws IOException {
+    void cleanupRelativePathsConvertAbsoluteToRelativePath(@TempDirectory.TempDir Path bibFolder) throws IOException {
         CleanupPreset preset = new CleanupPreset(CleanupPreset.CleanupStep.MAKE_PATHS_RELATIVE);
 
-        File tempFile = bibFolder.newFile();
+        Path path = bibFolder.resolve("AnotherRandomlyNamedFile");
+        Files.createFile(path);
+        File tempFile = path.toFile();
         BibEntry entry = new BibEntry();
         LinkedFile fileField = new LinkedFile("", tempFile.getAbsolutePath(), "");
         entry.setField("file", FileFieldWriter.getStringRepresentation(fileField));
 
         worker.cleanup(preset, entry);
         LinkedFile newFileField = new LinkedFile("", tempFile.getName(), "");
-        Assert.assertEquals(Optional.of(FileFieldWriter.getStringRepresentation(newFileField)), entry.getField("file"));
+        assertEquals(Optional.of(FileFieldWriter.getStringRepresentation(newFileField)), entry.getField("file"));
     }
 
     @Test
-    public void cleanupRenamePdfRenamesRelativeFile() throws IOException {
+    void cleanupRenamePdfRenamesRelativeFile(@TempDirectory.TempDir Path bibFolder) throws IOException {
         CleanupPreset preset = new CleanupPreset(CleanupPreset.CleanupStep.RENAME_PDF);
 
-        File tempFile = bibFolder.newFile();
+        Path path = bibFolder.resolve("AnotherRandomlyNamedFile");
+        Files.createFile(path);
+        File tempFile = path.toFile();
         BibEntry entry = new BibEntry();
         entry.setCiteKey("Toot");
         LinkedFile fileField = new LinkedFile("", tempFile.getAbsolutePath(), "");
         entry.setField("file", FileFieldWriter.getStringRepresentation(fileField));
 
         worker.cleanup(preset, entry);
         LinkedFile newFileField = new LinkedFile("", "Toot.tmp", "");
```

Comment:
If you use pdf as an extension above, then this should be also `Toot.pdf`. (The temp file rule always creates `tmp` files, thus the behavior before was correct).